scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -0,0 +1,1075 @@
+        # the parameter object. See "Creating a generator object" in
+        # http://statmath.wu.ac.at/software/unuran/doc/unuran.html#Concepts
+        self.par = NULL
+        if self.rng == NULL:
+            msg = self._messages.get()
+            if PyErr_Occurred():
+                return
+            raise UNURANError(msg)
+        unur_distr_free(self.distr)
+        self.distr = NULL
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    cdef inline void _rvs_cont(self, double[::1] out) except *:
+        """
+        Sample random variates from a continuous distribution.
+
+        Parameters
+        ----------
+        out : double[::1]
+            A memory view of size ``size`` to store the result.
+        """
+        cdef:
+            ccallback_t callback
+            unur_gen *rng = self.rng
+            size_t i
+            size_t size = len(out)
+
+        if self._callback_wrapper is not None:
+            init_unuran_callback(&callback, self._callback_wrapper,
+                                 self.params)
+        _lock.acquire()
+        try:
+            self._messages.clear()
+            unur_set_stream(self._messages.handle)
+
+            for i in range(size):
+                out[i] = unur_sample_cont(rng)
+                if PyErr_Occurred():
+                    return
+
+            msg = self._messages.get()
+            if msg:
+                raise UNURANError(msg)
+        finally:
+            _lock.release()
+            if self._callback_wrapper is not None:
+                release_unuran_callback(&callback)
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    cdef inline void _rvs_discr(self, int[::1] out) except *:
+        """
+        Sample random variates from a discrete distribution.
+
+        Parameters
+        ----------
+        out : int[::1]
+            A memory view of size ``size`` to store the result.
+        """
+        cdef:
+            ccallback_t callback
+            unur_gen *rng = self.rng
+            size_t i
+            size_t size = len(out)
+
+        if self._callback_wrapper is not None:
+            init_unuran_callback(&callback, self._callback_wrapper,
+                                 self.params)
+        _lock.acquire()
+        try:
+            self._messages.clear()
+            unur_set_stream(self._messages.handle)
+
+            for i in range(size):
+                out[i] = unur_sample_discr(rng)
+                if PyErr_Occurred():
+                    return
+
+            msg = self._messages.get()
+            if msg:
+                raise UNURANError(msg)
+        finally:
+            _lock.release()
+            if self._callback_wrapper is not None:
+                release_unuran_callback(&callback)
+
+    def rvs(self, size=None):
+        """
+        Sample from the distribution.
+
+        Parameters
+        ----------
+        size : int or tuple, optional
```

Comment:
To be linked with my comment on univariate distributions. It should be clear here that we are just resizing and not constructing some multivariate distribution.

Reply:
> It should be clear here that we are just resizing and not constructing some multivariate distribution.

Isn't that clear in the documentation and tutorial where I explicitly mention that the method works only for *univaraite* distributions?

Reply:
Right this just appears at the top of the sampling tutorial and in an exception. IMO it should be present in the note of the methods as not all will read this. Optional here.

Reply:
> Right this just appears at the top of the sampling tutorial and in an exception

I just noticed that I don't mention these methods are applicable only to univariate distributions in the class documentation. I think it is better to add this in the class documentation itself rather than a note here.

Reply:
Done. (in https://github.com/scipy/scipy/pull/14215#discussion_r681869291)